cmd/ctr/commands/run/run_unix.go

```diff
@@ -210,10 +249,51 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 }
 
 func getNewTaskOpts(context *cli.Context) []containerd.NewTaskOpts {
+	var (
+		tOpts []containerd.NewTaskOpts
+	)
 	if context.Bool("no-pivot") {
-		return []containerd.NewTaskOpts{containerd.WithNoPivotRoot}
+		tOpts = append(tOpts, containerd.WithNoPivotRoot)
+	}
+	if uidmap := context.String("uidmap"); uidmap != "" {
+		parts := strings.Split(uidmap, ":")
+		uid, _ := parseRemappingPart(parts[1], "uidmap", "uid")
+		tOpts = append(tOpts, containerd.WithUIDMap(uid))
+	}
+	if gidmap := context.String("gidmap"); gidmap != "" {
+		parts := strings.Split(gidmap, ":")
+		gid, _ := parseRemappingPart(parts[1], "gidmap", "gid")
+		tOpts = append(tOpts, containerd.WithGIDMap(gid))
+	}
+	return tOpts
+}
+
+func parseRemapping(remapping, idType string) (uint32, uint32, uint32, error) {
```

Comment:
just pass the input and return a `spec.LinuxIDMapping`; and handle error string creation by wrapping the generic "parsing failed" message from this function with the specifics of what you were trying to parse (uid or gid or whatever).

Reply:
Did as suggested!